tests/components/tplink/test_siren.py

```diff
@@ -0,0 +1,113 @@
+"""Tests for siren platform."""
+
+from __future__ import annotations
+
+from kasa import Device, Module
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.siren import (
+    DOMAIN as SIREN_DOMAIN,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+)
+from homeassistant.const import ATTR_ENTITY_ID, STATE_OFF, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+
+from . import (
+    DEVICE_ID,
+    _mocked_device,
+    _mocked_feature,
+    setup_platform_for_device,
+    snapshot_platform,
+)
+
+from tests.common import MockConfigEntry
+
+ENTITY_ID = "siren.hub"
+
+
+@pytest.fixture
+async def mocked_hub(hass: HomeAssistant) -> Device:
+    """Return mocked tplink hub with an alarm module and features."""
+
+    features = [
+        _mocked_feature(
+            "alarm",
+        ),
+        _mocked_feature(
+            "test_alarm",
+        ),
+        _mocked_feature(
+            "stop_alarm",
+        ),
+    ]
+
+    return _mocked_device(
+        alias="hub",
+        modules=[Module.Alarm],
+        features=features,
+        device_type=Device.Type.Hub,
+    )
+
+
+async def test_siren(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    entity_registry: er.EntityRegistry,
+    mocked_hub: Device,
+) -> None:
+    """Test initialization."""
+    await setup_platform_for_device(hass, mock_config_entry, Platform.SIREN, mocked_hub)
+
+    entity = entity_registry.async_get(ENTITY_ID)
+    assert entity
+    assert entity.unique_id == f"{DEVICE_ID}_siren"
+
+    state = hass.states.get(ENTITY_ID)
+    assert state.state == STATE_OFF
```

Comment:
This is kinda already tested by the `test_states` test

Reply:
Yeah, I think it makes no sense to do explicit testing in this case, so I'll just remove this test. Thanks for the review!